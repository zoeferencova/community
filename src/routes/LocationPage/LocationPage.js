import React, { Component } from "react";
import { Link } from "react-router-dom";
import CommUnityContext from "../../contexts/context";
import styles from "./LocationPage.module.css";
import GoogleMap from "../../components/GoogleMap/GoogleMap";
import MapSearch from "../../components/MapSearch/MapSearch"

export default class LocationPage extends Component {
    static contextType = CommUnityContext;

    state = {
        location: this.context.currentUser.location,
        radius: this.context.currentUser.radius
    }

    handleLocationChange = location => {
        this.setState({ location });
    }

    handleRadiusChange = e => {
        this.setState({ radius: e.target.value });
    }
    
    render() {
        return (   
            <main className={styles.main}>
                <h3>Location Settings</h3>
                <div className={styles.map}>
                    <GoogleMap radius={this.state.radius} location={this.state.location} displayMarker={true} />
                </div>
                <form className={styles.form}>
                    <div>
                        <label htmlFor="location">Location</label>
                        <MapSearch handleLocationChange={this.handleLocationChange} currentLocation={this.state.location} />
                    </div>
                    <div>
                        <label htmlFor="radius">Radius</label>
                        <input type="number" step="0.25" name="radius" id="radius" value={this.state.radius} onChange={this.handleRadiusChange} />
                        <span>miles</span>
                    </div>
                    <Link to="/home"><button>Cancel</button></Link>
                    <button type="submit">Use this location</button>
                </form>
            </main>
        )
    }
}