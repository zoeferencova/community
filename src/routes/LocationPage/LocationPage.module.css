.main {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 0 auto;
    position: absolute;
    top: 65px;
    padding: 38px;
    overflow-y: scroll
}

.form div {
    margin: 15px 0;
}

.map {
    height: 300px;
    width: 90%;
    margin-bottom: 15px;
}

@media only screen and (min-width: 600px) {
    .map {
        width: 400px;
    }
}