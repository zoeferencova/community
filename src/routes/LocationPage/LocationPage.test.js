import React from "react";
import { render } from "enzyme";
import LocationPage from "./LocationPage";
import { BrowserRouter } from "react-router-dom";
import CommUnityContext from "../../contexts/context";

import mockContext from "../../contexts/mock-context";

describe("LocationPage", () => {
    it("renders LocationPage component given mock context data", () => {
        const mockProps = {
            location: mockContext.user.location,
            radius: parseFloat(mockContext.user.radius)
        }

        const wrapper = render(
            <CommUnityContext.Provider value={mockContext}>
                <BrowserRouter>
                    <LocationPage {...mockProps} />
                </BrowserRouter>  
            </CommUnityContext.Provider>
        );
        
        expect(wrapper).toMatchSnapshot();
    });
});