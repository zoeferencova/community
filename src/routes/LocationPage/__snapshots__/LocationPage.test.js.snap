// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationPage renders LocationPage component given mock context data 1`] = `
<main
  class="main"
/>
`;
